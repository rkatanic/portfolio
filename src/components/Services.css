.service-card {
  position: relative;
}

.service-card-content {
  position: relative;
  padding: 2rem;
  margin: 2rem 0;
  border: 1px solid var(--color-primary);
  background-color: var(--color-body);
  overflow: hidden;
  cursor: pointer;
}

.service-card-content:hover .service-card-content-title::before {
  width: 3.5rem;
}

.service-card-icon {
  top: -0.75rem;
  left: 2rem;
  padding: 0 1rem;
  background-color: var(--color-body);
  position: absolute;
  z-index: 1;
}

.service-card-icon .icon {
  fill: var(--color-primary);
}

.service-card-content-badge {
  color: var(--color-primary);
}

.service-card-content-title {
  text-transform: uppercase;
  letter-spacing: 2px;
  position: relative;
}

.service-card-content-title::before {
  position: absolute;
  content: "";
  bottom: 0;
  width: 2rem;
  height: 2px;
  background-color: var(--color-primary);
  transition: 0.5s ease;
}

.service-card-content-desc {
  text-align: justify;
}

.service-card-binary {
  display: flex;
  width: 100%;
  height: 400px;
  overflow: hidden;
  transform: rotate(20deg);
  position: absolute;
  right: -70%;
  top: 0;
  opacity: 0.1;
  color: var(--color-primary);
}

.service-card-binary-text {
  margin: 0;
  padding: 0;
  cursor: pointer;
  writing-mode: vertical-rl;
  text-orientation: upright;
  position: relative;
  font-family: monospace;
  font-size: 1.125rem;
  margin-top: -50%;
}

.service-card-binary-text-up {
  animation: text-up 30s linear infinite;
}

.service-card-binary-text-down {
  animation: text-down 30s linear infinite;
}

@keyframes text-up {
  0% {
    top: 0;
  }
  50% {
    top: 50%;
  }
  100% {
    top: 0;
  }
}

@keyframes text-down {
  0% {
    top: 50%;
  }
  50% {
    top: 0;
  }
  100% {
    top: 50%;
  }
}
