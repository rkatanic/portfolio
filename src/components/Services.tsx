import { ReactComponent as CodeIcon } from "../assets/icons/code.svg";

import "./Services.css";

const Services = (): JSX.Element => (
  <>
    <h3 className="headline">Services I provide</h3>
    <div className="service-card">
      <div className="service-card-icon">
        <CodeIcon />
      </div>
      <div className="service-card-content">
        <h3 className="service-card-content-title">Front End Development</h3>
        <p className="service-card-content-desc">
          Using <b className="service-card-content-badge">HTML</b>,
          <b className="service-card-content-badge"> CSS</b>,
          <b className="service-card-content-badge"> Javascript</b>, and
          frameworks such as
          <b className="service-card-content-badge"> React</b> or
          <b className="service-card-content-badge"> Angular</b>, I can build
          fast, responsive and interactive client-side applications.
        </p>
        <div className="service-card-binary">
          <p className="service-card-binary-text service-card-binary-text-up">
            10100101000000010111111000001001010
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10010101010101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10001010100000001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10111010101010101010000100001000011
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10101010100000001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10000101110101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10101010111100001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10000101110101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10100101000000010111111000001001010
          </p>
        </div>
      </div>
    </div>
    <div className="service-card">
      <div className="service-card-icon">
        <CodeIcon />
      </div>
      <div className="service-card-content">
        <h3 className="service-card-content-title">Back End Development</h3>
        <p className="service-card-content-desc">
          Building scalable and maintainable server applications in
          <b className="service-card-content-badge"> Java</b>, using
          <b className="service-card-content-badge"> Spring Boot</b> framework
          and databases such as
          <b className="service-card-content-badge"> MySQL</b> or
          <b className="service-card-content-badge"> MongoDB</b>.
        </p>
        <div className="service-card-binary">
          <p className="service-card-binary-text service-card-binary-text-up">
            10100101000000010111111000001001010
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10010101010101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10001010100000001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10111010101010101010000100001000011
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10101010100000001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10000101110101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10101010111100001011111100000100101
          </p>
          <p className="service-card-binary-text service-card-binary-text-down">
            10000101110101010000100001000010101
          </p>
          <p className="service-card-binary-text service-card-binary-text-up">
            10100101000000010111111000001001010
          </p>
        </div>
      </div>
    </div>
  </>
);

export default Services;
