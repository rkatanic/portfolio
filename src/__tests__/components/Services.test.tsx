import { render } from "@testing-library/react";
import Services from "../../components/Services";

describe("Services", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Services />);

    expect(baseElement).toMatchSnapshot();
  });
});
