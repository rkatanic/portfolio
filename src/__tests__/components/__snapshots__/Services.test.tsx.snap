// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services should render 1`] = `
<body>
  <div>
    <h3
      class="headline"
    >
      Services I provide
    </h3>
    <div
      class="service-cards"
    >
      <div
        class="service-card"
      >
        <div
          class="service-card-content"
        >
          <h3
            class="card-title"
          >
            Front-End Development
          </h3>
          <p
            class="service-card-content-desc"
          >
            Using modern languages and frameworks, I can develop fast and responsive single-page client applications.
          </p>
          <div
            class="service-card-content-badges"
          >
            <b
              class="service-card-content-badge"
            >
              HTML
            </b>
            <b
              class="service-card-content-badge"
            >
              CSS
            </b>
            <b
              class="service-card-content-badge"
            >
              JavaScript
            </b>
            <b
              class="service-card-content-badge"
            >
              Sass
            </b>
            <b
              class="service-card-content-badge"
            >
              React
            </b>
            <b
              class="service-card-content-badge"
            >
              Angular
            </b>
          </div>
        </div>
      </div>
      <div
        class="service-card"
      >
        <div
          class="service-card-content"
        >
          <h3
            class="card-title"
          >
            Back-End Development
          </h3>
          <p
            class="service-card-content-desc"
          >
            I can develop scalable and maintainable server applications in Java, using Spring Boot framework.
          </p>
          <div
            class="service-card-content-badges"
          >
            <b
              class="service-card-content-badge"
            >
              Java
            </b>
            <b
              class="service-card-content-badge"
            >
              Spring Boot
            </b>
            <b
              class="service-card-content-badge"
            >
              MySQL
            </b>
            <b
              class="service-card-content-badge"
            >
              MongoDB
            </b>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
